README.md

```diff
@@ -207,6 +207,23 @@ func application(_ application: UIApplication, didFinishLaunchingWithOptions lau
 * Open `RInAppMessaging.xcworkspace` in Xcode then build/run
 * To run the tests press key shortcut command-U
 
+# **SDK Logic**
+
+## User cache
+
+Each user has a separate cache container that is persisted in UserDefaults. Each combination of rakutenId and userId is treated as a different user including a special - anonymous user - that represents non logged-in user (rakutenId and userId are null or empty).
+The key to access cache container is a SHA256 hash created from user identifiers and salt (which is a MD5 hash of user identifiers).
```

Comment:
I don't think an SDK consumer needs to know about our hashing. The rest of this section is fine.